src/helpers/ui/paste_interactive.ts

```diff
@@ -42,9 +45,35 @@ export function interactivePaste(
 export function interactivePasteFromOS(
   env: SpreadsheetChildEnv,
   target: Zone[],
-  text: string,
+  clipboardContent: ClipboardContent,
   pasteOption?: ClipboardPasteOptions
 ) {
-  const result = env.model.dispatch("PASTE_FROM_OS_CLIPBOARD", { target, text, pasteOption });
+  let result: DispatchResult;
+  try {
+    result = env.model.dispatch("PASTE_FROM_OS_CLIPBOARD", {
```

Comment:
it's worth mentioning in a comment that we do not trust the clipboard content to be sound and complete and that's why we try/catch to catch unexpected errors from a content that would be malformed for the current version.